[CIT-646] Add child device support doc (#594)
thin-edge/thin-edge.io: docs/src/architecture/mapper.md

After the mapper publishes a message on the topic `c8y/measurement/measurements/create`,
the message will be transferred to the topic `measurement/measurements/create` by [the MQTT bridge](../references/bridged-topics.md).

### For child devices

> The Cumulocity mapper supports sending measurements to child devices under the `thin-edge.io` parent device.

Reviewer: This sounds a bit incorrect / incomplete. I would say:

```
The Cumulocity mapper collects measurements not only from the main device but also from child devices.
These measurements are collected under the `tedge/measurements/<child-id>` topics
and forwarded to Cumulocity to corresponding child devices created under the `thin-edge.io` parent device.
```

The incoming topic is different from the parent device.
For child devices, use `tedge/measurements/<child-id>` topic. (`<child-id>` is your desired child device ID.)

The mapper works in the following steps.

> 1. When the mapper receives a Thin Edge JSON message on the `tedge/measurements/<child-id>`,

Reviewer: ```suggestion
1. When the mapper receives a Thin Edge JSON message on the `tedge/measurements/<child-id>` topic,
```

   the mapper publishes the SmartREST `101` message to create a child device under the `thin-edge.io` parent device.

>    The name of the child device is the given child ID, and the type is `thin-edge.io-child`.

Reviewer: ```suggestion
   The child device is named after the `<child-id>` topic name, and the type is `thin-edge.io-child`.
```

2. Publish corresponded Cumulocity JSON measurements messages over MQTT.

> 3. If the given child ID by the topic name is already known by the mapper,

Reviewer 2: That's an implementation detail and I'm not sure if this level of detail is required in the userdoc. For the user, the relevant thing is that "the child device is created on receipt of the very first measurement for that child device".

Reviewer: Agree with @albinsuresh . The user doc is not a specification. The proposed sentence is perfect.

   the mapper doesn't publish the first child device creation message again.
   This known child ID list is kept in memory.

If the incoming Thin Edge JSON message (published on `tedge/measurements/child1`) is as follows,

```json
{
  "temperature": 23
}
```

it gets translated into JSON via MQTT by the Cumulocity mapper.

```json
{
  "type":"ThinEdgeMeasurement",
  "externalSource":{
    "externalId":"child1",
    "type":"c8y_Serial"
  },
  "time":"2013-06-22T17:03:14+02:00",
  "temperature":{
    "temperature":{
      "value":23
    }
  }
}
```

## Azure IoT Hub mapper

> The Azure IoT Hub mapper takes messages formatted in the [Thin Edge JSON](thin-edge-json.md) as input.

Reviewer: Yes adding a single sentence to say that "child device measurements are not supported yet on Azure" would be good.
